File: plugin-dotnet-server/src/main/java/jetbrains/buildServer/dotnet/commands/Dotnet/TestCommandType.java
/*
 * Copyright 2000-2016 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * See LICENSE in the project root for license information.
 */

package jetbrains.buildServer.dotnet.commands.Dotnet;

import jetbrains.buildServer.dotnet.DotnetConstants;
import jetbrains.buildServer.dotnet.commands.CommandType;
import org.jetbrains.annotations.NotNull;

/**
 * Provides parameters for dotnet test command.
 */
public class TestCommandType implements CommandType {
    @NotNull
    @Override
    public String getName() {
        return DotnetConstants.COMMAND_TEST;
    }

    @NotNull
    @Override
    public String getEditPage() {
        return "editTestParameters.jsp";
    }

    @NotNull
    @Override
    public String getViewPage() {
        return "viewTestParameters.jsp";
    }
}

File: plugin-dotnet-server/src/main/kotlin/jetbrains/buildServer/dotnet/commands/TestCommandType.kt
package jetbrains.buildServer.dotnet.commands

import jetbrains.buildServer.dotnet.DotnetCommandType
import jetbrains.buildServer.dotnet.requirements.SDKBasedRequirementFactory

/**
 * Provides parameters for dotnet test command.
 */
class TestCommandType(
    sdkBasedRequirementFactory: SDKBasedRequirementFactory
) : DotnetCLICommandType(sdkBasedRequirementFactory) {
    override val name: String = DotnetCommandType.Test.id

    override val editPage: String = "editTestParameters.jsp"

    override val viewPage: String = "viewTestParameters.jsp"

    override fun validateProperties(properties: Map<String, String>) = sequence {
        yieldAll(super.validateProperties(properties))
        yieldAll(TestRetryCountValidator.validate(properties))
    }
}